watertap/unit_models/thickener.py

```diff
@@ -65,12 +67,127 @@ class ActivatedSludgeModelType(Enum):
+            overwrite=overwrite,
+        )
+        self.propagate_state_scaling(
+            target_state=model.underflow_state,
+            source_state=model.mixed_state,
+            overwrite=overwrite,
+        )
+        self.propagate_state_scaling(
+            target_state=model.overflow_state,
+            source_state=model.mixed_state,
+            overwrite=overwrite,
+        )
+
+        self.call_submodel_scaler_method(
+            submodel=model.underflow_state,
+            method="variable_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+        self.call_submodel_scaler_method(
+            submodel=model.overflow_state,
+            method="variable_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+
+        # Scale unit level variables
+        self.scale_variable_by_default(model.volume[0], overwrite=overwrite)
+        self.scale_variable_by_default(
+            model.electricity_consumption[0], overwrite=overwrite
+        )
+        self.scale_variable_by_default(model.diameter, overwrite=overwrite)
+        self.scale_variable_by_default(model.height, overwrite=overwrite)
+        self.scale_variable_by_default(
+            model.hydraulic_retention_time[0], overwrite=overwrite
+        )
+
+    def constraint_scaling_routine(
+        self, model, overwrite: bool = False, submodel_scalers: dict = None
+    ):
+        """
+        Routine to apply scaling factors to constraints in model.
+        Submodel Scalers are called for the property and reaction blocks. All other constraints
+        are scaled using the inverse maximum scheme.
+        Args:
+            model: model to be scaled
+            overwrite: whether to overwrite existing scaling factors
+            submodel_scalers: dict of Scalers to use for sub-models, keyed by submodel local name
+        Returns:
+            None
+        """
+        # Call scaling methods for sub-models
+        self.call_submodel_scaler_method(
+            submodel=model.mixed_state,
+            method="constraint_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+        self.call_submodel_scaler_method(
+            submodel=model.underflow_state,
+            method="constraint_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+        self.call_submodel_scaler_method(
+            submodel=model.overflow_state,
+            method="constraint_scaling_routine",
+            submodel_scalers=submodel_scalers,
+            overwrite=overwrite,
+        )
+
+        # Scale unit level constraints
+        for c in model.component_data_objects(Constraint, descend_into=True):
+            self.scale_constraint_by_nominal_value(
+                c,
+                scheme=ConstraintScalingScheme.inverseMaximum,
```

Comment:
This is the part that could potentially be informed by ScalingProfiler in particular